### PR TITLE
Allow InterviewStatus to be triggered by Scheduling

### LINKED ISSUE #106: Allow `InterviewStatus` to be triggered by scheduling
Allow the `InterviewStatus` to be automatically changed when user schedules an interview. When user schedules an interview, `InterviewStatus` will change from `Not Scheduled` to `Scheduled`

### PR DESCRIPTION
**Pull request by tiewweijian**
- Currently the InterviewStatus has to be triggered manually
- In this iteration, the InterviewStatus will be triggered when a user schedules an interview
- With this, the InterviewStatus will not be able to be edited manually in this iteration as well.

closes #106

### CODE REVIEW

**Comment by charmainehly** [comment]
Closes #106  ! - not sure if @tiewweijian you'll need to add this to the main PR comment in order to link it.

**Comment by tiewweijian** [comment]
@charmainehly oh ya thanks for adding in ! i was trying to find the issue number, then I forgot cause I wanted to resolve the conflicts first ! But thanks for helping me to add in !

**Review by domlimm — COMMENTED** [review]
Nice job wj! Just some minor queries. Will leave it as Comment for now

File: src/main/java/seedu/address/model/interview/Interview.java [hunk 1]
```diff
@@ -24,11 +24,14 @@
      * Every field must be present and not null.
      */
     public Interview(Candidate candidate, LocalDateTime interviewDateTime) {
+        Candidate candidate1;
```
[the remark this answers is not in the document]
Comment (tiewweijian) [reply src/main/java/seedu/address/model/interview/Interview.java]: Removed it in the new PR. Thanks for pointing it out !

**Review by lzan98 — APPROVED** [review]
LGTM! 👍

**Review by domlimm — APPROVED** [review]
LGTM! Good job sir!

**Pull request merged**